### DISCUSSION TITLE
Requests from server-side are many times slower than from the client-side

### DISCUSSION THREAD

**Discussion by marco-streng** [vercel/next.js#22778] [Help]
Currently I'm developing an app which uses AWS Cognito for authentication and AWS AppSync as a GraphQL server.

I like to protect routes on the server side, so I implement a quite simple call to AWS Cognito (like shown in the AWS tutorials)

```javascript
import { withSSRContext } from 'aws-amplify'

export async function getServerSideProps ({ req }) {
  const { Auth } = withSSRContext({ req })

  try {
    const user = await Auth.currentAuthenticatedUser()

    return {
      props: {
        authenticated: true,
        user: user.username
      }
    }
  } catch (err) {
    return {
      redirect: {
        destination: '/login'
      }
    }
  }
}
```

**The Problem**

I've deployed the app to Vercel (production mode) and wondered about the slow performance (high FCP). After some research I found out, that this request takes about **400-500ms** on server-side.
But the same request done on the client-side run's in less than **15ms**.

The same difference appears when I want fetch data from the GraphQL endpoint (ServerSide: ~900ms, ClientSide: ~150ms)

At first I thought on a geographical distance Issue. But my AWS Recources are all deployed in eu-central-1 (Frankfurt, Germany) and the Vercel functions in AWS eu-west-3 (Paris, France). In other words: There is no atlantic ocean in between.

Maybe someone has got an idea or made a similar experience?
Thanks a lot!

**Comment by ErnestBrandi** [comment]
Hi, 

- have you tried with a totally different API endpoint and noticing the same difference ? 
- how do you compute time ?
- do you account the fact that with SSR you might also compute the rendering of the page ?
- then have you tried this comparison on another page of your website ?

**Reply by marco-streng** [reply]
Hi,

thanks for your answer @ErnestBrandi.

I measure the duration by comparing unix timestamps right before and after the request runs.
I've tried two other API endpoints (GitHub API and dev.to API) and in this case, there was no huge difference between ServerSide and ClientSide at all. So I think it's an issue on the Amplify side and I will do more research here.

Thanks!

**Comment by lilienblum** [comment]
Hi @marco-streng,

I noticed the same thing. API (AppSync) requests from getServerSideProps with `withSSRContext` are 2-3 times slower than from the client-side `import { API } from 'aws-amplify'`.  I'm getting 400-600ms currently and it is unacceptable.
Did you manage to solve this?

FYI opened a ticket: https://github.com/aws-amplify/amplify-js/issues/8751

**Reply by marco-streng** [reply]
Hi @dn-l 

unfortunately I haven't been able to find a solution so far. I had to move all requests to the client.

Thanks for opening the ticket!

**Reply by lilienblum** [reply]
Gotya! If you don't mind can you leave a comment in the ticket so they know I'm not the only one with the problem

**Reply by marco-streng** [reply]
I've just read that you were able to solve this issue @dn-l.
Maybe you can explain your solution? Thx!

**Reply by lilienblum** [reply]
@marco-streng I didn’t change anything. It’s still slow on localhost but pretty quick on aws (after cold start), 100-200ms.

**Reply by lilienblum** [reply]
It just started to work faster with no reason in my case
